pandas/io/stata.py

```diff
@@ -1141,72 +1246,136 @@ def _read_strls(self):
             if self.path_or_buf.read(3) != b'GSO':
                 break
 
-            v_o = struct.unpack(self.byteorder + 'L',
-                                self.path_or_buf.read(8))[0]
-            typ = self.path_or_buf.read(1)
+            v_o = struct.unpack(self.byteorder + 'Q', self.path_or_buf.read(8))[0]
+            typ = struct.unpack('B', self.path_or_buf.read(1))[0]
             length = struct.unpack(self.byteorder + 'I',
                                    self.path_or_buf.read(4))[0]
-            self.GSO[v_o] = self.path_or_buf.read(length-1)
-            self.path_or_buf.read(1)  # zero-termination
+            va = self.path_or_buf.read(length)
+            if typ == 130:
+                va = va[0:-1].decode(self._encoding or self._default_encoding)
+            self.GSO[v_o] = va
 
+    # legacy
+    @Appender('DEPRECATED: ' + _data_method_doc)
```

Comment:
> how old is this method, anyone know?

It is the original method from the initial import AFAICT.  The statsmodel's code tha reads stata data files has the same structure (they clearly have the same origina).  I suspect that `read()` is rarely called since almost everything can be done using `from_stata`.